core/src/main/java/org/wso2/msf4j/PersistentSessionManager.java

```diff
@@ -94,15 +106,37 @@ public void saveSession(Session session) {
     }
 
     @Override
-    public void deleteSession(Session session) {
-        String pathname = Paths.get(SESSION_DIR, session.getId()).toString();
+    public void deleteSession(Session session, MicroServiceContext microServiceContext) {
+        String microServiceSessionDir = microServiceContext.getService().getClass().getName();
+        String pathname = Paths.get(SESSION_DIR + "/" + microServiceSessionDir, session.getId()).toString();
```

Comment:
Can we get rid of "/" here and let the Paths API construct the OS dependent path?